py_trees/tests.py

```diff
@@ -27,12 +31,24 @@ def pre_tick_visitor(behaviour_tree):
     print("\n--------- Run %s ---------\n" % behaviour_tree.count)
 
 
-def tick_tree(tree, visitor, from_iteration, to_iteration):
-    print("\n================== Iteration %s-%s ==================\n" % (from_iteration, to_iteration))
-    for i in range(from_iteration, to_iteration + 1):
+def tick_tree(tree,
+              from_tick,
+              to_tick,
+              visitor=None,
+              print_snapshot=False,
+              print_blackboard=False
+              ):
+    print("\n================== Iteration {}-{} ==================\n".format(from_tick, to_tick))
+    for i in range(from_tick, to_tick + 1):
         print(("\n--------- Run %s ---------\n" % i))
         for node in tree.tick():
-            node.visit(visitor)
+            if visitor is not None:
+                node.visit(visitor)
+    if print_snapshot:
+        print(console.green + "\nAscii Tree Snapshot" + console.reset)
+        display.print_ascii_tree(tree, show_status=True)
+    if print_blackboard:
+        print(str(blackboard.Blackboard()))
```

Comment:
Should these prints be withing the for loop so we get snapshot for every tick?

Reply:
@stonier just this thing and approved.

Reply:
It can be very useful for debugging the test to see the snapshot at every tick. By default it's off though since most cases don't need it.